### PR TITLE
Add instructions to configure code style and checkstyle for IDEA

### PR DESCRIPTION
**Pull request by cbornet**
### Modifications

* Added sections in the README

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

This change is a trivial rework / code cleanup without any test coverage.

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (yes / no) no
  - The public API: (yes / no) no
  - The schema: (yes / no / don't know) no
  - The default values of configurations: (yes / no) no
  - The wire protocol: (yes / no) no
  - The rest endpoints: (yes / no) no
  - The admin cli options: (yes / no) no
  - Anything that affects deployment: (yes / no / don't know) no

### Documentation

  - Does this pull request introduce a new feature? (yes / no) no
  - If yes, how is the feature documented? (not applicable / docs / JavaDocs / not documented)
  - If a feature is not applicable for documentation, explain why?
  - If a feature is not documented yet in this PR, please create a followup issue for adding the documentation

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -211,6 +211,39 @@ required plugins.
...
+3. Set **Checkstyle version** to **8.37**
+
+4. Set **Scan scope** to **Only Java sources (including tests)**
+
+5. Click the **+** button in the **Configuration** section to open a dialog to choose the checkfile file.
+   1. Enter a **Description**. Eg. Pulsar.
+   2. Select **Use a local checkstyle file**.
+   3. Set **File** to **buildtools/src/main/resources/pulsar/checkstyle.xml**.
+   4. Select **Store relative to project location**.
+   5. Click **Next**, then **Next** again, then **Finish**.
```
Comment (lhotari) [inline README.md:241]: it seems that the step for selecting the `buildtools/src/main/resources/pulsar/suppressions.xml` file is missing. IIRC, there's a dialog where that is requested.

**Review by lhotari — APPROVED** [review]

File: README.md (hunk 1 above)
Comment (cbornet) [reply README.md:241]: `suppressions.xml` is loaded by default. There's a property to use another filename but it's not needed for us.

File: README.md [hunk 2]
```diff
@@ -211,6 +211,39 @@ required plugins.
...
+
+3. Pick the file `${pulsar_dir}/src/idea-code-style.xml`
+
+4. On the dialog box that opens, click **OK**.
+
+5. Ensure the scheme you just created is selected in **Scheme** dropdown then click **OK**.
+
+#### Configure Checkstyle
+
+1. Install the Checkstyle-IDEA plugin
```
Comment (Anonymitaet) [inline README.md:228]:
```suggestion
1. Install the Checkstyle-IDEA plugin.
```

File: README.md [hunk 3]
```diff
@@ -211,6 +211,39 @@ required plugins.
...
+
+2. Open Checkstyle Settings dialog box by going to **Settings** -> **Tools** -> **Checkstyle**.
+
+3. Set **Checkstyle version** to **8.37**
```
Comment (Anonymitaet) [inline README.md:232]:
```suggestion
3. Set **Checkstyle version** to **8.37**.
```

File: README.md [hunk 4]
```diff
@@ -211,6 +211,39 @@ required plugins.
...
+
+4. Set **Scan scope** to **Only Java sources (including tests)**
```
Comment (Anonymitaet) [inline README.md:234]:
```suggestion
4. Set **Scan scope** to **Only Java sources (including tests)**.
```

File: README.md [hunk 5]
```diff
@@ -211,6 +211,39 @@ required plugins.
...
+
+5. Click the **+** button in the **Configuration** section to open a dialog to choose the checkfile file.
```
Comment (Anonymitaet) [inline README.md:236]:
```suggestion
5. Click **+** button in the **Configuration** section to open a dialog to choose the checkfile file.
```

File: README.md [hunk 6]
```diff
@@ -211,6 +211,39 @@ required plugins.
...
+   1. Enter a **Description**. Eg. Pulsar.
```
Comment (Anonymitaet) [inline README.md:237]:
```suggestion
   1. Enter a **Description**. For example, Pulsar.
```

File: README.md (hunk 1 above)
Comment (Anonymitaet) [inline README.md:241]:
```suggestion
   5. Click **Next** -> **Next** -> **Finish**.
```

**Review by Anonymitaet — COMMENTED** [review]
Many thanks for your contribution. I’ve left some comments, PTAL, thanks.

**Review by Anonymitaet — APPROVED** [review]

**Pull request merged**